Mathlib/Dynamics/PeriodicPts.lean

```diff
@@ -628,42 +630,109 @@ namespace MulAction
 
 open Function
 
-variable {α β : Type*} [Group α] [MulAction α β] {a : α} {b : β}
+universe u v
+variable {α : Type v}
+variable {G : Type u} [Group G] [MulAction G α]
+variable {M : Type u} [Monoid M] [MulAction M α]
+
+/--
+The period of a multiplicative action of `g` on `a` is the smallest `n` such that `g ^ n • a = a`,
+or `0` if such an `n` does not exist.
+-/
+@[to_additive "The period of an additive action of `g` on `a` is the smallest `n`
+such that `(n • g) +ᵥ a = a`, or `0` if such an `n` does not exist."]
+noncomputable def period (m : M) (a : α) : ℕ := minimalPeriod (fun x => m • x) a
+
```

Comment:
I think we should add a few more lemmas about special cases of `period`, in particular `period_one : period 1 x = 1` and `period_inv : period g⁻¹ x = period g x`? (Assuming those are correct!)